### ISSUE TITLE
Add repos by orgs. Select org first?

### ISSUE THREAD

**Issue by dhruvagarwal** [middlewarehq/middleware#172]
### Description:

A user might be a part of multiple orgs and while selecting the repositories, it becomes confusing to find the right org. 

### Expected behavior:

It would be great to choose the orgs first. This will also help to reduce the clutter in the Add Repos option.

### Actual behavior:
Today, it loads up a list of repos across my personal and from other orgs.
<img width="1130" alt="Screenshot 2024-04-24 at 7 12 13 PM" src="https://github.com/middlewarehq/middleware/assets/4587641/ec1eccd7-6cf0-43fe-bcdb-8599177c6f21">

**Comment by e-for-eshaan** [comment]
does #445 solve this?

**Comment by dhruvagarwal** [comment]
I don't think so. Adding repos by org might require user to repeatedly add org name for search.

**Comment by e-for-eshaan** [comment]
noted!

**Comment by hardikbhatia777** [comment]
Hey, I'd be happy to take this up.

If it's possible to strictly enforce the flow of choosing the org before the repo, I think we can change the `useReposSearch` method in `useTeamsConfig.tsx` to only search for orgs, and using that value, render a similar searchbox on the side with the repositories owned by that org (perhaps rename things to useOrgsSearch and useOrgReposSearch?)

This will obviously not let the users search repos directly though. Maybe keep the current box as it is, and alongside it, provide this flow as well? Let me know which implementation you'd prefer.

**Comment by jayantbh** [comment]
Hey @hardikbhatia777, we're picking this up internally actually. Apologies this was unassigned.

However, as a good first issue, you could [pick up this one](https://github.com/middlewarehq/middleware/issues/478) if you're interested.

**Comment by jayantbh** [comment]
Ah I see. Tbh it was picked but not really solved. The associated pr may be closed without merging so you're free to give it a shot anyway.
